snapcraft/plugins/python.py

```diff
@@ -0,0 +1,254 @@
+      (list)
+      A list of dependencies to get from PyPi
+    - python-version:
+      (string; default: python3)
+      The python version to use. Valid options are: python2 and python3
+"""
+
+import os
+import re
+import stat
+import subprocess
+from glob import glob
+
+import snapcraft
+from snapcraft import file_utils
+
+
+class PythonPlugin(snapcraft.BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+        schema['properties']['requirements'] = {
+            'type': 'string',
+        }
+        schema['properties']['constraints'] = {
+            'type': 'string',
+        }
+        schema['properties']['python-packages'] = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'default': [],
+        }
+        schema['properties']['process-dependency-links'] = {
+            'type': 'boolean',
+            'default': False,
+        }
+        schema['properties']['python-version'] = {
+            'type': 'string',
+            'default': 'python3',
+            'enum': ['python2', 'python3']
+        }
+        schema.pop('required')
+
+        # Inform Snapcraft of the properties associated with pulling. If these
+        # change in the YAML Snapcraft will consider the pull step dirty.
+        schema['pull-properties'].extend([
+            'requirements',
+            'constraints',
+            'python-packages',
+            'python-version',
+        ])
+
+        return schema
+
+    @property
+    def plugin_build_packages(self):
+        if self.options.python_version == 'python3':
+            return [
+                'python3-dev',
+                'python3-pip',
+                'python3-pkg-resources',
+                'python3-setuptools',
+            ]
+        elif self.options.python_version == 'python2':
+            return [
+                'python-dev',
+                'python-pip',
+                'python-pkg-resources',
+                'python-setuptools',
+            ]
+
+    @property
+    def plugin_stage_packages(self):
+        if self.options.python_version == 'python3':
+            return ['python3']
+        elif self.options.python_version == 'python2':
+            return ['python']
+
+    @property
+    def system_pip_command(self):
+        if self.options.python_version == 'python3':
+            return os.path.join(os.path.sep, 'usr', 'bin', 'pip3')
+        else:
+            return os.path.join(os.path.sep, 'usr', 'bin', 'pip')
+
+    def __init__(self, name, options, project):
+        super().__init__(name, options, project)
+        self.build_packages.extend(self.plugin_build_packages)
+        self.stage_packages.extend(self.plugin_stage_packages)
+
+    def env(self, root):
+        return [
+            'PYTHONUSERBASE={}'.format(root),
+            'PYTHONHOME={}'.format(os.path.join(root, 'usr'))
```

Comment:
Take a look, this is where I see the problem:
```
$ ls prime/usr/lib/python2.7/dist-packages/
README
$ ls prime/usr/lib/python2.7/site-packages
ls: cannot access 'prime/usr/lib/python2.7/site-packages': No such file or directory
$ ls prime/lib/python2.7/site-packages/
argparse-1.2.2.dist-info  easy_install.py   functools32-3.2.3.post2.dist-info  pip                  setuptools                   wheel-0.29.0.dist-info
argparse.py               easy_install.pyc  jsonschema                         pip-8.1.2.dist-info  setuptools-26.1.1.dist-info
argparse.pyc              functools32       jsonschema-2.5.1.dist-info         pkg_resources        wheel
```

So there's nothing in usr, all the installed packages end in lib. What I don't yet understand is why when I try with jsonschema, it finds the module anyway. But when I try with my matrix snap, it doesn't. :pensive: